src/pybind/mgr/dashboard/frontend/src/app/ceph/new-dashboard/dashboard/dashboard.component.ts

```diff
@@ -37,4 +85,50 @@ export class DashboardComponent implements OnInit {
         version[0] + ' ' + version.slice(2, version.length).join(' ');
     });
   }
+
+  getDf() {
+    this.clusterService.getCapacity().subscribe((data: any) => {
+      this.dfData = data;
+    });
+  }
+
+  prepareRawUsage(chart: Record<string, any>, data: Record<string, any>) {
+    const nearFullRatioPercent = this.osdSettings.nearfull_ratio * 100;
+    const fullRatioPercent = this.osdSettings.full_ratio * 100;
+    const percentAvailable = this.calcPercentage(
+      data.stats.total_bytes - data.stats.total_used_raw_bytes,
+      data.stats.total_bytes
+    );
+    const percentUsed = this.calcPercentage(
+      data.stats.total_used_raw_bytes,
+      data.stats.total_bytes
+    );
+    if (percentUsed >= fullRatioPercent) {
+      this.color = 'chart-color-red';
+    } else if (percentUsed >= nearFullRatioPercent) {
+      this.color = 'chart-color-yellow';
+    } else {
+      this.color = 'chart-color-blue';
+    }
+
+    chart.dataset[0].data = [
+      Math.round(nearFullRatioPercent),
+      Math.round(Math.abs(nearFullRatioPercent - fullRatioPercent)),
+      Math.round(100 - fullRatioPercent)
+    ];
+
+    chart.dataset[1].data = [percentUsed, percentAvailable];
+    chart.dataset[1].backgroundColor[0] = this.cssHelper.propertyValue(this.color);
+
+    chart.dataset[0].label = [
+      `${percentUsed}%\nof ${this.dimlessBinary.transform(data.stats.total_bytes)}`
```

Comment:
Here you need [a bit of magic](https://stackoverflow.com/a/52934578), since JS doesn't provide anything like Python `str.format` (late interpolation). I'm sure there'll be some library that provides Python's str.format behaviour, but if you don't want to explore, something like this should work:
```suggestion
      (data: IData) => `${data.current/data.total}%\nof ${this.dimlessBinary.transform(data.total)}`
```

Reply:
You're right! I didn't see that this function was called from the on changes hook too.